### DISCUSSION TITLE
Announcing Video DownloadHelper v8.2 Beta

### DISCUSSION THREAD

**Discussion by paulrouget** [aclap-dev/video-downloadhelper#13] [Announcements]
We're excited to announce Video DownloadHelper version: v8.2 Beta.

For now, version 8.2 is only available on the Beta Channel (more info about the beta channel here: https://github.com/aclap-dev/video-downloadhelper/wiki/Beta-versions/).

The Addon Beta is immediately available for Firefox (you can force the update in the Firefox Settings). And is still in the review queue for Chrome and Edge.

### New forum

Until now, we have been using Google Group as our main way to communicate with our users, and we are now experimenting with Github Discussions & Wiki. We can discuss issues, questions and suggestions there, and find documentation in the wiki. The Google Group is not going away, but we have experienced too many issues with this system and believed it's not future-proof.

Looking forward talking with all of you here :)

### SideDownload becomes the default download method.

Last update introduced the SideDownload method. We are happy to report that this new download mechanism has significantly improved the quality of the downloads. We are now making it the default download method, and simply renamed it "Download".

### New "Download Audio Only" method

A new action is now available to only download the audio track of a video. Pretty amazing to download a Youtube playlist as a collection of MP3 :)

### Better MPD streams support

We have significantly improved MPD downloads (mostly noticeable on Vimeo). This requires a Companion App upgrade, but we will only request upgrade if you attempt to download MPD content (not very common).

### Smarter video stream detection

The stream detection mechanism has been improved. You can tweak its behavior in the settings: `Settings > More > Video Qualities`.

### Updated locales

Special thanks to our contributors! You too can help improve Video DownloadHelper by contributing to our locales on Github or Google Group
https://github.com/aclap-dev/video-downloadhelper/discussions/categories/language-translation

### Road to 8.3

The versions 8.1 and 8.2 are mostly about stability, modernisation of the internals, and introducing the new side download mechanism.

8.3 will focus on the general user experience and the interface. If you have suggestions about how to make Video DownloadHelper better, please post on the Github forum :)

**Comment by GrampaWildWilly** [comment]
When I first installed the beta, my keyboard shortcut for dropping down the VDH menu didn't work.  So I closed all browser windows then relaunched Firefox & that restored the keyboard shortcut.  So anybody installing this beta probably will want to cycle the browser right after installing.  This beta still correctly recognizes my CoApp 2.0.14.  It also recognizes my existing license without my having to do anything about it.  This is on Windows 11 Pro 64-bit 23H2.

**Comment by GrampaWildWilly** [comment]
Actually, reloading the extension may have been enough.  But I didn't think of that at the time.  You might just do that instead & see if that's enough.  If not, then cycling the browser is in order.

**Comment by GrampaWildWilly** [comment]
I debated whether to post this as a new topic but I think this is a one-time issue connected with your first use of the beta.  This is what happened to me.  I don't assume it will happen to everybody else, but it might happen to somebody else, so here's my warning.  When you do your first download with this new beta, do NOT assume that your default action is still in effect.  My experience was that my first download launched a Quick Download, something I almost never do.  My previously set default action was Download, & that's not what my first download did.  So for you first download, use the secondary VDH menu & set or reset your default action.  From that point forward, you should have no issues.

**Comment by thelwyn** [comment]
Hi,

Thank you for this update it solved a problem I had with videos in a private website!

However I can't use the "Assemble local audio and video files" function any more. Nothing happens when I click on the button (icon looks like 2 lines merging into one). If I click "Convert local files", the usual window opens though.

My information:

Video DownloadHelper 8.2.0.17a1 (beta)
Langue du navigateur: fr
Généré le 2024-01-30
Options de génération: browser=firefox, linuxlic=false, noyt=false, target=mozilla
Plateforme Win x86-64
Navigateur Mozilla Firefox 122.0
Appli compagnon trouvée: VdhCoApp 2.0.14
Binaire de l'appli compagnon: C:\Program Files\DownloadHelper CoApp\vdhcoapp.exe

Thanks in advance.

**Reply by paulrouget** [reply]
Ah good catch! Will look into it.

**Reply by paulrouget** [reply]
@thelwyn mind me asking, what do you use the merge local function for? I'm asking because, normally, this function should be unnecessary with the 8.2 download system (as in… you should not be in a situation where a video is being downloaded without its audio).

**Reply by thelwyn** [reply]
@paulrouget before this update I had a lot of trouble with video from a private subscription website, it would download video and sound but video would freeze immediately. The workaround I found at that time was to demux it using another small soft based on ffmpeg, and then merge it again through VDH function. I have a backlog of previously downloaded video that I need to fix like this ^^

**Reply by paulrouget** [reply]
@thelwyn hey - can you update the beta (v8.2.0.19a1) and see if merge works now? It will most likely ask you to update your coapp (sorry). The workflow has changed a bit (especially the video/audio file selection system).

**Reply by GrampaWildWilly** [reply]
Is 2.0.15 still the latest CoApp?  There's no release numbers shown here https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-Installation so there's no telling when a new one has appeared unless you post about it.

**Reply by GrampaWildWilly** [reply]
Never mind.  That page gives you CoApp 2.0.17 now.

**Reply by GrampaWildWilly** [reply]
The instructions on that page say you have to close all browser windows, then install the CoApp, only then can you reopen the browser.  I am finding that it's enough to go into the browser settings, disable VDH, run the CoApp installer, then just enable VDH again, leaving all your browser windows open the whole time.  Of course, disabling VDH does kill any inflight downloads, so keep that in mind.

**Reply by GrampaWildWilly** [reply]
Which reinforces the importance of noting the current release number of the CoApp on that page.  If you wait to let VDH prompt you to get a new CoApp, you're still going to have to at least disable VDH before you can install the CoApp.  Seems like a bit of an after you, no after you situation.  Tell us what the latest CoApp release is on that page so we can gracefully shut off VDH, get & install a new CoApp that might have appeared, then enable VDH again.  The prompt is nice in VDH but it would be nicer to see the CoApp release number on that Wiki page.

**Reply by paulrouget** [reply]
Sorry that's not gonna happen.

**Reply by thelwyn** [reply]
@paulrouget it's working again, thank you!

**Comment by GrampaWildWilly** [comment]
This isn't an answer to the question you actually asked, which is why doesn't the Aggregate local audio and video files function launch when you click the button.  It doesn't launch for me, either.  But if you need to aggregate files, it is really simple to do using ffmpeg on your own, without using VDH.  First, you have to get one of the ffmpeg packages from ffmpeg.org.  I think they have installers.  I always just get the zip file & unzip it somewhere on my disk space.  I always get the package marked "full"  just to be sure I'm getting the full-featured version.  I feel like it's better to have more than you need but not use it all than to have not enough & discover you need more.

Once you have ffmpeg on your system, you can execute a command at a command prompt like this:

ffmpeg -i "video track" -i "audio track" -codec: copy "aggregated file" 2>"ffmpeg log file"

You would replace the first 3 items in double quotation marks with full file specifications of the form
x:\dir\dir\dir\filename.mp4.  I'm saying mp4 but it could be one of the other file types, whatever you have.  The last item in double quotation marks is the full file specification for the ffmpeg log file.  The 2> thing causes Windows to redirect the output of the ffmpeg command to a file, where you can read it at your leisure using your favorite text editor.  Without the 2> thing, the results of the command will fly by in the command window faster than any human could possibly read them.  Be sure you include the double quotation marks surrounding your file specifications.  You would do well to write this command in a .bat file so you can execute the command again without retyping it all if you encounter any errors.

**Reply by thelwyn** [reply]
Thanks @GrampaWildWilly.

**Comment by paulrouget** [comment]
I'm working on fixing merging & converting this week. A new update will hopefully fix that soon.

**Comment by harapra** [comment]
HI, i just installed this version on firefox122 running on Ubuntu22.04.3. Unfortunately the VideoDownloadHelper icon does not display in its usual position. Am I missing something? Easy to get back? Thanks

**Reply by paulrouget** [reply]
Does that help?

https://github.com/aclap-dev/video-downloadhelper/wiki/Lost-icon

**Reply by harapra** [reply]
Now I feel stupid, but yes it did help. I have it back. Maybe it would be a good idea to make this behavior, i. e. pin it, default.
Thank you very much.

**Reply by paulrouget** [reply]
Sadly we don't have control over that :(

**Comment by whocrazy** [comment]
Hello. How do I force the update to Beta? I can't find anywhere in my firefox settings that will let me do this.  The only update options available are default, on and off.
am I missing something?

**Reply by jcfrog** [reply]
If should be able to ask for update in extensions panel (Ctrl + Shift + A)
<img width="969" alt="Screenshot 2024-02-04 at 08 41 43" src="https://github.com/aclap-dev/video-downloadhelper/assets/368367/f536d020-3981-43ac-b015-c7a16789baab">

**Reply by GrampaWildWilly** [reply]
Oh yeah.  I didn't interpret his words that way.  But what you're showing will cause Firefox to check for updates to every add-on you have.  To check for an update to only VDH, start by doing this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/41e83913-7895-4aae-bd40-40f4d45c455c)

.
.
That will open a different display:
.
.

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/383f67fb-94c6-4f02-a2c1-74b595635899)

.
.
Then keep a sharp eye on this:
.
.

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/efb94ec6-86b4-4f76-bddd-82d26a8110c9)

**Comment by whocrazy** [comment]
I'm also having issues with one particular video site, some hls streams on the boxcast video platform error out with a "side download error.  Even when downloading them with plain old ffmpeg, sometimes the download will just freeze for about 5 minutes, then continue

**Comment by GrampaWildWilly** [comment]
To get on the beta channel, do what Paul says in the original post upthread here & click this:

https://github.com/aclap-dev/video-downloadhelper/wiki/Beta-versions

For your other issue, you'd probably be better off opening a new discussion.  Do be sure to supply a FULL description of  your environment including **release numbers** of all the relevant bits of software plus at least one **URL** of something that's giving you trouble.

**Comment by whocrazy** [comment]
Hi there.
I am totally blind and I use a screen reader called NVDA, so I am not able to see what is in those PNGs, sorry, You'll have to describe what to do.
Sorry about that.

**Reply by GrampaWildWilly** [reply]
Oh my . . .  Let me try my best . . .

In the Firefox settings on the add-ons page, which you access by visiting about:addons (treat it like a URL), there is a list of every add-on you have installed.  With each add-on there is a slider button that you can push left or right to disable or enable that particular add-on.  To the right of the slider there's a button that looks like 3 dots.  When you click that, it drops down a menu with some selections on it.  The bottom entry on that menu says Manage.  You're supposed to click that.  I'm trying to imagine how you do this without seeing it.  I salute your tenacity.

So once you click Manage, it takes you to a new display within the Firefox settings.  On this page there's a heading part way down the page for Allow automatic updates.  On this line you can select a policy for the browser to control how it processes updates automatically to this add-on.  The choices are Default, On, Off.  Because the browser is brain dead & does not check whether you are downloading something when it decides to automatically update VDH (or any extension, not just VDH), you should always set this option to Off.  You should never let your browser automatically update VDH because that could interrupt a download you might be doing.  Some users have reported exactly that occurrence & it justifiably upset them.  But that's how it works.  Slightly inconvenient, I admit, but the browser behavior forces this upon us.  You must manually check for updates to VDH.

You do that check by clicking the Check for Updates button on this same line.  When you click this button, there may or may not be an update available.  When there is an update available, you need to look at another one of those buttons with 3 dots.  Further up this page, there is a duplication of the slider for enabling/disabling the add-on, with another of those buttons with 3 dots.  If Check for Updates finds an update, a blue dot appears within this button with the 3 dots.  At that point, you need to click the 3 dots button.  That drops down a menu in which there will be an Update selection.  You need to click that.  Confirmation that an update occurred appears lower down the page where a version number is displayed.

You know, I'm writing this & I'm suddenly realizing that this is a lot of jumping around when you can't see the page.  Everything I've described is easy to see at a glance.  But if you can't see, this has to be an ordeal.  I'm so glad you're participating here.  Your contribution should make the rest of us think twice about how we do things.

**Reply by whocrazy** [reply]
Hey there.  Thanks for that in depth Description.
Yeah, I was thrown by the fact that updates were set to default and there was no visible way to check for updates, I didn't realise you had to turn that option off before the "check for updates" option became available.
Thanks for clearing that up.
Fortunately, firefox has built in support for screen readers, and most things are labeled so that I don't have to worry about clicking on the wrong option, especially in the addons section.
